app/Services/FilterService.php

```diff
@@ -135,6 +135,38 @@ protected function prepareFilters(array $availableFilters = []): self
                 unset($this->data['tags']);
             }
         }
+        // If we have data but no "locations" array, it's empty
```

Comment:
Please make a function instead of copy-pasting 6 lines of code and just changing one string